### PR TITLE
Vlussenburg cleanup and supervisord extensibility

### PR DESCRIPTION
**Pull request by vlussenburg**
@jdewinne please take a look - added the supervisor *.conf directive and did some image-size optimizations and cleanups

### CODE REVIEW

**Comment by vlussenburg** [comment]
@jdewinne I'm considering to actually have the supervisor.conf to only by the include of the *.conf and no actual programs. That makes it easier for extenders to replace how XLD is started without overriding the entire file. Maybe over engineering. Thoughts?

File: resources/supervisord.conf [hunk 1]
```diff
@@ -10,3 +10,6 @@ stdout_logfile=/dev/stdout
 stdout_logfile_maxbytes=0
 stderr_logfile=/dev/stderr
 stderr_logfile_maxbytes=0
+
+[include]
+files = /etc/supervisor/conf.d/*.conf
```
Comment (jdewinne) [inline resources/supervisord.conf]: On `alpine` the path `/etc/supervisor/conf.d/*.conf` doesn't exist by default. Not sure about `slim`.
So what happens if it doesn't exist? Does supervisor fail? And should we put this in the readme that you can add extra supervisor configuration files on this path?

Comment (vlussenburg) [reply resources/supervisord.conf]: Good point, and I did consider that. Actually that only results in a WARN and when you do a COPY of a file to that path no errors are given, so I thought a mkdir -p would be superfluous. That would be addressed if I actually make a separate xld.conf, which I'm even more leaning towards now ;)

Comment (vlussenburg) [reply resources/supervisord.conf]: Re: documentation on the fragment dir: that is a great point.

Comment (jdewinne) [reply resources/supervisord.conf]: Makes sense. Because a separate `xld.conf` would also get rid of the warning. You wanna make the change before merge?

Comment (vlussenburg) [reply resources/supervisord.conf]: Docs added. I'll make the change.

**Comment by vlussenburg** [comment]
@jdewinne should be all complete now.

**Pull request merged**